python/lsst/ts/salobj/sal_info.py

```diff
@@ -235,31 +245,41 @@ def parse_idl(self):
         * event_names: a tuple of event names, without the ``"logevent_"``
           prefix
         * telemetry_names: a tuple of telemetry topic names
+        * sal_topic_names: a tuple of SAL topic names, e.g.
+          "logevent_summaryState", in alphabetical order
         * revnames: a dict of topic name: name_revision
         """
-        pattern = re.compile(r" *struct +(?P<name_rev>(?P<name>.+)_(?:[a-zA-Z0-9]+)) +{ *")
+        struct_pattern = re.compile(r"\s*struct\s+(?P<name_rev>(?P<name>.+)_(?:[a-zA-Z0-9]+)) +{")
+        index_pattern = re.compile(rf"\s*long\s+{self.name}ID;")
         command_names = []
         event_names = []
         telemetry_names = []
+        sal_topic_names = []
         revnames = {}
+        self.indexed = False
```

Comment:
Good point. And in the doc string I added `indexed` to the list of attributes set by this method.

I confess I wish all SAL components were indexed (even those for which we always use index=0); it would eliminate a special case.